lib/cards/base.rb

```diff
@@ -0,0 +1,56 @@
+# frozen_string_literal: true
+
+module Cards
+  class Base
+    NUMBER_LENGTH = 16
+
+    attr_accessor :balance
+    attr_reader :number, :taxes
+
+    class << self
+      include VIEW
+
+      def generate_number
+        Array.new(NUMBER_LENGTH) { rand(10) }.join
+      end
+
+      def update_balance(card, amount, tax, operation:)
+        operation == :withdraw ? card.balance -= (amount - tax) : card.balance += (amount - tax)
+      end
+    end
+
+    def initialize(_balance)
+      @number = self.class.generate_number
+    end
+
+    private
+
+    def calculate_tax(amount, percent_tax, fixed_tax)
+      (amount / 100) * percent_tax + fixed_tax
+    end
+
+    def put_fixed
+      0
+    end
+
+    def put_percent
+      0
+    end
+
+    def withdraw_fixed
+      0
+    end
+
+    def withdraw_percent
+      0
+    end
+
+    def send_fixed
+      0
+    end
+
+    def send_percent
+      0
+    end
```

Comment:
In all cards classes move numbers to constants and use methods to return constants. Mentor told me to do so, it makes code more readable